### PR TITLE
Use list picker when searching a single value

### LINKED ISSUE #49279: Search Filters are buggy and annoying to work with in 51
### Describe the bug

The usability of a Text Filter set to a search box is very annoying and also buggy when setup to pick a single value:

Image

### To Reproduce

1. Go to Admin -> Table Metadata -> Products -> Find the Title column and set it as a search box:

Image

2. Create a question based on products and then link a test filter to it making sure the Filter is set to a search box and a single Value

3. Notice how it allows the addition of multiple values but then removes them. Also the list hides the Add Filter button from being selected and you have to click between the border to be able to go back to the Add Filter button. Check the video here:

https://github.com/user-attachments/assets/1907e139-577c-4b42-be2a-1ae59506006f

This is how it used to be on 50

https://github.com/user-attachments/assets/dc3d1d82-399a-4805-b1ca-952b899d1413

### Expected behavior

I want to be able to use the filter properly and not having to click between borders and also that if i set the filter to select single values then only 1 value is selected.

So there are currently 2 issues with this filter both around usability. 

### Logs

None that are relevant since it's a UI flow (flaw)

### Information about your Metabase installation

version 51.1

### Severity

Makes usability of Search Filters very annoying

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/49279

Switch back to the list filter instead of using the multi autocomplete input in dashboard filters.

### To verify

1. Go to Admin -> Table Metadata -> Products -> Title column and set it as Cateog
2. Create a dashboard that has a question relating to Products on it
3. Add a filter that is linked to the Title column
4. Set the filter type to `Search box` and the `A single value`
5. Try using the filter, and make sure it's using the list selector not the MultiAutoComplete 
input
5. This should also work when the filter type is set `Multiple values`

Screenshot 2024-10-30 at 17 13 28
Screenshot 2024-10-30 at 17 13 33

### CODE REVIEW

File: frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx [hunk 1]
```diff
@@ -163,10 +163,15 @@ export function FieldValuesWidgetInner({
   const [isExpanded, setIsExpanded] = useState(false);
   const dispatch = useDispatch();
 
+  const isSingleValueSearch = valuesMode === "search" && !multi;
```
Comment (kamilmielnik) [inline frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx:166]: Should we take `disableSearch` into account here?

```suggestion
  const isSingleValueSearch = valuesMode === "search" && !multi && !disableSearch;
```

**Review by kamilmielnik — APPROVED** [review]
Works as expected :+1: 

----

> Closes https://github.com/metabase/metabase/issues/49279

That issue mentions 2 problems (see "Expected behavior" section).

> I want to be able to use the filter properly and not having to click between borders 

Let's report this UX issue separately?

**Comment by romeovs** [comment]
@kamilmielnik 
> I want to be able to use the filter properly and not having to click between borders

This PR makes it so that the `MultiAutocomplete` is not used anymore so that issue is sidestepped completely.

**Comment by kamilmielnik** [comment]
> This PR makes it so that the `MultiAutocomplete` is not used anymore so that issue is sidestepped completely.

The issue with Dropdown covering the "Add filter" / "Update filter" button is still there in some other cases, e.g.

1. Map "Products" > "Title" to "Search box" in Table Metadata in Admin
2. In dashboard use Text > Contains > Search box > Multiple values filter and map it to Products > Title
3. Save dashboard
4. Use the filter, type a letter and wait for dropdown to show up

**Comment by romeovs** [comment]
> The issue with Dropdown covering the "Add filter" / "Update filter" button is still there in some other cases

@kamilmielnik I'm not seeing that behaviour is this branch locally.

https://github.com/user-attachments/assets/bea2f36a-1f20-48e3-aeff-150d82aef86c

Additionally this would not be as much of an issue since with any filter other than `Is` there is more space to click outside of the dropdown due to the header.

**Comment by romeovs** [comment]
@kamilmielnik You're right of course (and I was testing agains the wrong database locally, so not seeing the changes here). I've added some extra padding to the autocomplete component as discussed [in this slack thread](https://metaboat.slack.com/archives/C07BWMG9C8M/p1730207740154939?thread_ts=1730204267.569459&cid=C07BWMG9C8M).

Screenshot 2024-10-31 at 22 11 11

It's not ideal, but unblocks this blocker. I have ideas for better ways of doing this but they need to be run by product first.

**Pull request merged**